.github/workflows/CI.yml

```diff
@@ -180,3 +180,51 @@ jobs:
           name: prebuilds
           pattern: prebuilds-*
           delete-merged: true
+  smoke-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
+        pm:
+          - npm
+          - yarn
+          - pnpm
+    steps:
+      - uses: actions/setup-node@v4
+      - run: |
+          npm init --init-module testproject -y
```

Comment:
You can use this template package add an example script to it
https://github.com/zeromq/zeromq.js/tree/master/test/typings-compatibility/template